ioctl/cmd/bc/bcbucketlist.go

```diff
@@ -73,12 +86,54 @@ func (m *bucketlistMessage) String() string {
 }
 
 // getBucketList get bucket list from chain
-func getBucketList(arg string) error {
-	address, err := util.GetAddress(arg)
+func getBucketList(method string, args ...string) (err error) {
+	switch method {
+	case bucketlistMethodByVoter:
+		addr, offset, limit := "", uint64(0), uint64(1000)
+		if len(args) > 0 {
+			addr = args[0]
+		}
+		if len(args) > 1 {
+			offset, err = strconv.ParseUint(args[1], 10, 64)
+			if err != nil {
+				return output.NewError(output.ValidationError, "invalid offset", err)
+			}
+		}
+		if len(args) > 2 {
+			limit, err = strconv.ParseUint(args[2], 10, 64)
+			if err != nil {
+				return output.NewError(output.ValidationError, "invalid limit", err)
+			}
+		}
+		return getBucketListByVoter(addr, uint32(offset), uint32(limit))
+	case bucketlistMethodByCandidate:
```

Comment:
line 110~126 is duplicate code of 92~108

Reply:
so could be moved to before switch

Reply:
I parse args for specific method cuz consider another function `getBucketListByIndex(index uint32)`, but it's not import, I agree with you.

Reply:
we already have `ioctl bc bucket 36` to get bucket 36
unless you plan on  `ioctl bc bucketlist index [36, 37, 38]` for more than 1 bucket

Reply:
in that case, it is method = "index"